src/node.cpp

```diff
@@ -835,6 +877,25 @@ NodeCore::~NodeCore()
     delete attrstring;
 }
 
+PublicLink::PublicLink(PublicLink *plink)
+{
+    this->ph = plink->ph;
+    this->ets = plink->ets;
+    this->takendown = plink->takendown;
+}
+
+bool PublicLink::isExpired()
+{
+    if (!ets)       // permanent link: ets=0
+        return false;
+
+    time_t t = time(NULL);
+//    struct tm *ptm = gmtime(&t);
```

Comment:
I'd keep it there until the expiration feature is live and usable by client apps, so we can test if they are really not mandatory.

The documentation for the `time()` function says:
"The value returned generally represents the number of seconds since 00:00 hours, Jan 1, 1970 UTC (i.e., the current unix timestamp). Although libraries may use a different representation of time: Portable programs should not use the value returned by this function directly, but always rely on calls to other elements of the standard library to translate them to portable types (such as localtime, gmtime or difftime)."